src/NorthwindWebsite/Middleware/ImageCachingMiddleware.cs

```diff
@@ -0,0 +1,97 @@
+﻿using NorthwindWebsite.Business.Services.Interfaces;
+using NorthwindWebsite.Core.ApplicationSettings;
+using NorthwindWebsite.Core.Constants;
+using NorthwindWebsite.Core.Utils;
+
+namespace NorthwindWebsite.Middleware;
+
+public class ImageCachingMiddleware : IMiddleware
+{
+    private const string ImagesUrlSubstring = "/images/";
+
+    private readonly AppSettings _appSettings;
+    private readonly IImageCachingService _imageCachingService;
+    private readonly ILogger<ImageCachingMiddleware> _logger;
+
+    public ImageCachingMiddleware(
+        AppSettings appSettings,
+        IImageCachingService cacheService,
+        ILogger<ImageCachingMiddleware> logger)
+    {
+        _appSettings = appSettings;
+        _imageCachingService = cacheService;
+        _logger = logger;
+    }
+
+    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+    {
+        DateTime.TryParse(context.Response.Headers["expires"], out var expires);
```

Comment:
What will happen if I am a "great hacker" and will replace it with some string or invalid date format? Let's also verify that parsing was successfull